new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDomainOnPV.java

```diff
@@ -0,0 +1,706 @@
+   * Copies the WLST domain script to a temp location.
+   * Creates a domain properties in the temp location.
+   * Creates a configmap containing domain scripts and property files.
+   * Runs a job to create domain on persistent volume.
+   * @throws IOException when reading/writing domain scripts fails
+   */
+  private void createDomainOnPV() throws IOException {
+
+    logger.info("create a staging location for domain creation scripts");
+    Path pvTemp = Paths.get(RESULTS_ROOT, "ItDomainOnPV", "domainCreateTempPV");
+    FileUtils.deleteDirectory(pvTemp.toFile());
+    Files.createDirectories(pvTemp);
+
+    logger.info("copy the create domain WLST script to staging location");
+    Path srcWlstScript = Paths.get(RESOURCE_DIR, "python-scripts", "wlst-create-domain-onpv.py");
+    Path targetWlstScript = Paths.get(pvTemp.toString(), "create-domain.py");
+    Files.copy(srcWlstScript, targetWlstScript, StandardCopyOption.REPLACE_EXISTING);
+
+    logger.info("create WebLogic domain properties file");
+    Path domainPropertiesFile = Paths.get(pvTemp.toString(), "domain.properties");
+    assertDoesNotThrow(
+        () -> createDomainProperties(domainPropertiesFile), "Creating domain properties file failed");
+
+    logger.info("add files to a config map for domain creation job");
+    List<Path> domainScriptFiles = new ArrayList<>();
+    domainScriptFiles.add(targetWlstScript);
+    domainScriptFiles.add(domainPropertiesFile);
+
+    logger.info("Create a config map to hold domain creation scripts");
+    String domainScriptConfigMapName = "create-domain-scripts-cm";
+    assertDoesNotThrow(
+        () -> createConfigMapForDomainCreation(domainScriptConfigMapName, domainScriptFiles),
+        "Creating configmap for domain creation failed");
+
+    logger.info("Running a Kubernetes job to create the domain");
+    runCreateDomainJob(pvName, pvcName, domainScriptConfigMapName, domainNamespace);
+
+  }
+
+  /**
+   * Create a properties file for WebLogic domain configuration.
+   * @param wlstPropertiesFile path of the properties file
+   * @throws FileNotFoundException when properties file path not found
+   * @throws IOException when writing properties fails
+   */
+  private void createDomainProperties(Path wlstPropertiesFile) throws FileNotFoundException, IOException {
+    // create a list of properties for the WebLogic domain configuration
+    Properties p = new Properties();
+
+    p.setProperty("domain_path", "/shared/domains");
+    p.setProperty("domain_name", domainUid);
+    p.setProperty("cluster_name", clusterName);
+    p.setProperty("admin_server_name", adminServerName);
+    p.setProperty("managed_server_port", "8001");
+    p.setProperty("admin_server_port", "7001");
+    p.setProperty("admin_username", "system");
+    p.setProperty("admin_password", "gumby1234");
+    p.setProperty("admin_t3_public_address", K8S_NODEPORT_HOST);
+    p.setProperty("admin_t3_channel_port", "32001");
```

Comment:
when create domain custom resource, you used:
             .addChannelsItem(new Channel()
                         .channelName("T3Channel")
                         .nodePort(0))))
Do you want to hard code the admin_t3_channel_port here?

Reply:
The 32001 port is passed to domain creation scripts and used internally.
The 
addChannelsItem(new Channel()
.channelName("T3Channel")
.nodePort(0))
simply exposes that 32001 with available port in the node

Reply:
that won't work - the t3 port MUST be the same as the node port, or t3 will not connect